platform-controller/src/main/java/org/hobbit/controller/kubernetes/K8sContainerManagerImpl.java

```diff
@@ -0,0 +1,474 @@
+        try {
+            JobList services = k8sClient.batch().jobs().inNamespace("default").withLabel(label, value).list();
+            return services.getItems();
+        }catch (Exception e){
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public String getContainerId(String name) {
+        return name;
+    }
+
+    @Override
+    public String getContainerName(String containerId) {
+        return containerId;
+    }
+
+    @Override
+    public void addContainerObserver(ContainerStateObserver containerObserver) {
+        containerObservers.add(containerObserver);
+    }
+
+    @Override
+    public void pullImage(String imageName) {
+
+    }
+
+    @Override
+    public PodMetrics getStats(String containerId) {
+        PodMetrics stats = null;
+        try {
+            stats = k8sClient.top().pods().metrics("default", containerId);;
+        } catch (Exception e) {
+            LOGGER.warn("Error while requesting usage stats for {}. Returning null. Error: {}", containerId,
+                e.getLocalizedMessage());
+        }
+        return stats;
+    }
+
+    private String getInstanceName(String imageName) {
+        return getInstanceName(imageName, "");
+    }
+
+    private String getInstanceName(String imageName, String prefix) {
+        String baseName = imageName;
+        // If there is a tag it has to be removed
+        if (containsVersionTag(baseName)) {
+            int pos = imageName.lastIndexOf(':');
+            baseName = baseName.substring(0, pos - 1);
+        }
+        int posSlash = baseName.lastIndexOf('/');
+        int posColon = baseName.lastIndexOf(':');
+        if (posSlash > posColon) {
+            baseName = baseName.substring(posSlash + 1);
+        } else if (posSlash < posColon) {
+            baseName = baseName.substring(posColon + 1);
+        }
+        final String uuid = UUID.randomUUID().toString().replaceAll("-", "");
+        StringBuilder builder = new StringBuilder(prefix.length() + baseName.length() + uuid.length() + 2);
+        if (prefix.length() != 0) {
+            builder.append(prefix);
+            builder.append('-');
+        }
+        builder.append(baseName.replaceAll("[^-a-z0-9]", "-"));
+        int maxLength = DOCKER_MAX_NAME_LENGTH - 1 - uuid.length();
+        if (builder.length() > maxLength) {
+            builder.setLength(maxLength);
+        }
+        builder.append('-');
+        builder.append(uuid);
+        return builder.toString();
+    }
+
+    public static boolean containsVersionTag(String imageName) {
+        int pos = 0;
+        // Check whether the given image name contains a host with a port
+        Matcher matcher = PORT_PATTERN.matcher(imageName);
+        while (matcher.find()) {
+            pos = matcher.end();
+        }
+        // Check whether there is a ':' in the remaining part of the image name
+        return imageName.indexOf(':', pos) >= 0;
+    }
+}
```

Comment:
This method, as well as many others, should be in a shared parent/abstract class since it's identical to the one in `ContainerManagerImpl`.